text/0043-block-metadata.md

```diff
@@ -0,0 +1,338 @@
+and those auditing it. The following information is proposed to be distributed
+on the blockchain itself:
+1. An AVR that proves a block was signed by an SGX enclave signed by the Intel chain
+of trust
+2. Block Metadata including the node's quorum set, AVR, and SCP message signing key 
+signed by the SCP message signing key to prove that the node was the holder of that
+private key when the block was signed
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+When consensus nodes externalize a block, they will include `BlockMetadata` with
+the AVR and quorum set for that block.
+
+LedgerDB and WatcherDB also track and persist the block metadata.
+
+Note that while the `BlockMetadata` can be used to verify chain integrity, it
+couldn't be used to falsify any new transactions or blocks.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+Add a `message BlockMetadata` to `blockchain.ArchiveBlock`, and an analogous
+`struct BlockMetadata` to `mc_blockchain_types::BlockData`.
+
+## Protobuf schema
+```protobuf
+message BlockMetadataContents {
+    /// The Block ID.
+    BlockID id = 1;
+
+    /// Quorum set configuration at the time of externalization.
+    QuorumSet quorum_set = 2;
+
+    /// IAS report for the enclave which generated the signature.
+    VerificationReport verification_report = 3;
+}
+
+message BlockMetadata {
+    /// Metadata signed by the consensus node.
+    BlockMetadataContents contents = 1;
+
+    /// Message signing key.
+    Ed25519Public node_key = 2;
+
+    /// Signature using `node_key` over the Digestible encoding of `contents`.
+    Ed25519Signature signature = 3;
+}
+
+// NB: Only addition is field #4.
+message ArchiveBlockV1 {
+    /// Block.
+    Block block = 1;
+
+    /// Contents of the block.
+    BlockContents block_contents = 2;
+
+    /// Block signature.
+    BlockSignature signature = 3;
+
+    /// Additional signed metadata about this block.
+    BlockMetadata metadata = 4;
+}
+```
+
+Since this is a protobuf-additive change, we don't need a new `ArchiveBlock`
+version/type.
+
+## Signing metadata
+The metadata signature will always differ from the block signature. The block
+signature is generated by the enclave using a private key that never leaves the
+enclave, while the metadata is signed by the "untrusted" code outside the
+enclave using the SCP message signing key, since the enclave does not have this
+metadata.
+
+The metadata signature is generated from the `Digestible` encoding of
+`BlockMetadataContents`, since the protobuf wire format is not a canonical
+encoding.
+
+## Signing key history
+MobileCoin Foundation will publish a mapping/list of node_id to a range of block
+indexes, which node operators and consumers can use to verify the metadata
+signatures.
+
+This mapping will contain multiple node_id:block_range entries, in the formats defined below.
+
+#### Key History TOML format
+```toml
+[[node]]
```

Comment:
It might be worth including the responder id here - at the very least it would make it easier for humans to figure out where each key is supposed to be coming from.

Reply:
I hesitate to add a field we cannot use (because there's no way to get a responder ID from any block data); humans can include the responder ID in a comment.

Reply:
Whats the difference between having it here and having it in the `AVR History` file?